source/extensions/filters/http/cache/cache_filter.h

```diff
@@ -45,16 +45,27 @@ enum class FilterState {
   Destroyed
 };
 
+class CacheFilterConfig : public Router::RouteSpecificFilterConfig {
+public:
+  CacheFilterConfig(std::shared_ptr<HttpCache> http_cache,
```

Comment:
How would CacheFilterConfig know enough about the extension to be able to sanitize it?

getCache gets its information from the pre-sanitized config because the extension is the only thing that can possibly know how to sanitize that part of the config, and CacheFilter itself can't have any direct dependency on what's in "the mystery bag".

It would be feasible to make CacheFilterConfig's constructor handle getCache, rather than doing that first and passing it in, but getCache can potentially throw a bad config exception, and my understanding is that throwing exceptions during constructors is problematic.

Reply:
Certainly CacheFilterConfig can't sanitize CacheConfig's typed_config().value(), as it's opaque.

The class name "CacheFilterConfig" tells me that it's a class for holding the CacheFilter's config. std::shared_ptr<HttpCache> is not part of CacheFilter's config, so it shouldn't be part of a class named CacheFilterConfig.

There's nothing wrong with throwing exceptions during constructors--that's their original use-case. However, it's unclear whether the current getCache interface can survive if we need to support per-route CacheConfig::typed_config.